library/src/main/java/kwasm/format/text/Limits.kt

```diff
@@ -37,12 +36,9 @@ fun List<Token>.parseLimits(startingIndex: Int): ParseResult<Limit> {
     )
     min.magnitude = 32
     val maxOrCloseParenIndex = startingIndex + 1
-    if (maxOrCloseParenIndex >= this.size || this[maxOrCloseParenIndex] is Paren.Closed)
+    if (maxOrCloseParenIndex >= this.size || this[maxOrCloseParenIndex] !is IntegerLiteral.Unsigned)
         return ParseResult(Limit(min.value.toUInt(), UInt.MAX_VALUE), 1)
-    val max = this[maxOrCloseParenIndex] as? IntegerLiteral.Unsigned ?: throw ParseException(
-        "Expected integer literal",
-        this[maxOrCloseParenIndex].context
-    )
+    val max = (this[maxOrCloseParenIndex] as IntegerLiteral.Unsigned)
```

Comment:
I've noticed it's a bit safer to use `getOrNull(index)` rather than array-index location syntax. This way you just get a null Token, as opposed to causing an OutOfBoundException.  It might be worth fixing that wherever else it's used here.